### DISCUSSION TITLE
Keep track of your wanted music (discussion/brainstorming)

### DISCUSSION THREAD

**Discussion by sindreruud** [beetbox/beets#4780] [General]
Hi!

One thing I'm struggling with is keeping track of music I want to add to my collection, managed by Beets. Generally I find music on Spotify (playlists, and suggestion algorithm), YouTube, Soundcloud, Discogs and Bandcamp. I will usually add this music/release to a specific "wanted" playlist, or favourite it or whatever means the service has to save it. Then at some point I will go through these lists to acquire the music and then run it through beets to add it to my collection.

So my first question is: Does this workflow sound familiar to anyone? Do you have any specific tools or apps you use to make this process better? Please share!

The dream scenario I envision is some kind of small web app that basically can fetch the data from various services (if they have an API), and then organize the releases/tracks (metadata, not the actual releases/tracks) in the web app. You could even pre-fetch metadata from musicbrainz/discogs with Beets, and ofcoure check if you maybe have it already. 

I know Spotify has an open API that you could use to fetch tracks from a specific playlist, for instance. 

Curious to hear your setups and if you have any input! :)

**Comment by arsaboo** [comment]
Discovery is the primary reason that I invest time with Beets. There are still a few gaps, but over time I have developed a reasonable routine to surface good music. 

I almost exclusively use Plexamp to listen to my music and keep rating tracks there. I use the `plexsync` function to keep the plex ratings synced in Beets. For the tracks that are unrated, I rely on the Spotify `popularity` field to identify tracks that may be worthy of attention. Beyond the popularity, I rely on curated playlists (suggestions/recommendations) across various services to identify upcoming/popular tracks. I use the `plexplaylistimport` feature in my [Plexsync](https://github.com/arsaboo/beets-plexsync) plugin to bring them into Plex (in a temporary playlist, which I can then listen and rate). The `plexplaylistimport` works with Apple, Spotify, JioSaavn, and Gaana.com currently (and can be easily extended to add other services) and it lists the tracks that I don't have in my library. 

Once I have all the information in beets, I can use `plexplaylistadd` to create playlists in Plex based on beets queries (e.g., `popularity` > 50 and a specific `genre`). I have cron jobs set for many of these things to automate everything. 

There are a few holes in my workflow that need to be plugged...but overall I am reasonably happy with my setup. One of the things that I would love to have is some sort of [content recommendation](https://github.com/beetbox/beets/discussions/4689).   

I would love to hear what others are doing for discovery.

**Comment by gryphonmyers** [comment]
> Hi!
> 
> One thing I'm struggling with is keeping track of music I want to add to my collection, managed by Beets. Generally I find music on Spotify (playlists, and suggestion algorithm), YouTube, Soundcloud, Discogs and Bandcamp. I will usually add this music/release to a specific "wanted" playlist, or favourite it or whatever means the service has to save it. Then at some point I will go through these lists to acquire the music and then run it through beets to add it to my collection.
> 
> So my first question is: Does this workflow sound familiar to anyone? Do you have any specific tools or apps you use to make this process better? Please share!
> 
> The dream scenario I envision is some kind of small web app that basically can fetch the data from various services (if they have an API), and then organize the releases/tracks (metadata, not the actual releases/tracks) in the web app. You could even pre-fetch metadata from musicbrainz/discogs with Beets, and ofcoure check if you maybe have it already.
> 
> I know Spotify has an open API that you could use to fetch tracks from a specific playlist, for instance.
> 
> Curious to hear your setups and if you have any input! :)

The closest thing I know of to what you're describing is https://lidarr.audio/. Lidarr is _okay_ but very much wants you to do things its way. This means it doesn't integrate very well with a Beets library, mainly because Lidarr is very hands on about file management (something Beets obviously also does). The best you can do is use something like https://github.com/rafaelparente/beets-lidarr-fields to get the file naming conventions to line up, then try to trigger beets to import albums when they appear in Lidarr.  I spent some time trying to set this up, but decided it was more trouble than it was worth. 

I just wrote this plugin to do what I need: https://github.com/gryphonmyers/beets-wanted-albums-plugin It doesn't currently pull wanted albums from streaming services (something you described yourself doing) - currently it just lets you add artists you like on the CLI and it will let you run a command for each album you're missing by that artist.
